toolkit/tools/pkg/rpmssnapshot/rpmssnapshot.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
This shares a lot of functionality with specreader, which is in dire need of an update. Might be a good future todo item to take the stuff here and replace the old specreader code with a common version.

Reply:
Right again. I even tried to re-use some of the specreader's code but its chroot setup was slightly different than the one I needed here, so refactoring would have been too robust for this PR.

What's more important, though, is that I'm waiting for #2983 to make it into `main`. I see it as sort of more thorough groundwork enabling us to re-use code more easily. I didn't want to make too many changes to existing code, which would make updating that large rework even more difficult.